scripts/end_to_end_test.py

```diff
@@ -1,22 +1,357 @@
+                    logger.info(f"{status} {result.module_name} ({result.execution_time:.2f}s)")
+                except Exception as e:
+                    logger.error(f"Test execution failed for {test_case.module_name}: {e}")
+                    self.results.append(
+                        WorkflowTestResult(
+                            module_name=test_case.module_name, success=False, execution_time=0.0, error=str(e)
+                        )
+                    )
+
+        return self.results
+
+    def generate_report(self) -> Dict[str, Any]:
+        """Generate a comprehensive test report"""
+        total_tests = len(self.results)
+        successful_tests = sum(1 for r in self.results if r.success)
+        failed_tests = total_tests - successful_tests
+
+        total_time = sum(r.execution_time for r in self.results)
+        avg_score = sum(r.score for r in self.results) / total_tests if total_tests > 0 else 0.0
+
+        report = {
+            "summary": {
+                "total_tests": total_tests,
+                "successful_tests": successful_tests,
+                "failed_tests": failed_tests,
+                "success_rate": successful_tests / total_tests if total_tests > 0 else 0.0,
+                "total_execution_time": total_time,
+                "average_score": avg_score,
+            },
+            "results": [asdict(result) for result in self.results],
+        }
+
+        return report
+
+    def print_summary_table(self):
+        """Print a formatted summary table to console"""
+        print("\n" + "=" * 80)
+        print("WORKFLOW E2E TEST RESULTS")
+        print("=" * 80)
+
+        print(f"{'Workflow':<40} {'Status':<10} {'Time':<10} {'Score':<10}")
+        print("-" * 80)
+
+        for result in sorted(self.results, key=lambda x: x.module_name):
+            status = "PASS" if result.success else "FAIL"
+            time_str = f"{result.execution_time:.2f}s"
+            score_str = f"{result.score:.2f}" if result.success else "N/A"
+
+            print(f"{result.module_name:<40} {status:<10} {time_str:<10} {score_str:<10}")
+
+        total_tests = len(self.results)
+        successful_tests = sum(1 for r in self.results if r.success)
+        success_rate = successful_tests / total_tests if total_tests > 0 else 0.0
+
+        print("-" * 80)
+        print(f"Total: {total_tests}, Passed: {successful_tests}, Failed: {total_tests - successful_tests}")
+        print(f"Success Rate: {success_rate:.1%}")
+        print("=" * 80)
+
 
 def main():
-    """
-    Main entry point for end-to-end testing.
+    """Main entry point for the E2E test script"""
+    import argparse
+
+    parser = argparse.ArgumentParser(description="Run end-to-end tests for Vellum Workflows SDK")
+    parser.add_argument("--max-workers", type=int, default=4, help="Maximum number of concurrent workers")
+    parser.add_argument("--filter", type=str, help="Filter workflows by name substring")
```

Comment:
Update our release yaml to pass in a `--filter skip-all` argument for now, while we work out the kinks

Reply:
The release yaml should also define the following environment variable:
```
VELLUM_API_URL=http://localhost:8000
```